flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/DefaultSlotPoolServiceSchedulerFactory.java

```diff
@@ -193,4 +197,37 @@ public static DefaultSlotPoolServiceSchedulerFactory fromConfiguration(
         return new DefaultSlotPoolServiceSchedulerFactory(
                 slotPoolServiceFactory, schedulerNGFactory);
     }
+
+    private static AdaptiveSchedulerFactory getAdaptiveSchedulerFactoryFromConfiguration(
+            Configuration configuration) {
+        Duration initialResourceAllocationTimeout =
+                returnValueOrReplaceDefaultIfReactiveMode(
+                        configuration,
+                        JobManagerOptions.RESOURCE_WAIT_TIMEOUT,
+                        Duration.ofMillis(-1));
+        Duration resourceStabilizationTimeout =
+                returnValueOrReplaceDefaultIfReactiveMode(
+                        configuration,
+                        JobManagerOptions.RESOURCE_STABILIZATION_TIMEOUT,
+                        Duration.ZERO);
```

Comment:
Would it be simpler to define the defaults based on `configuration.get(JobManagerOptions.SCHEDULER_MODE)` here and then simply call `configuration.getOptional(option).orElseGet(() -> defaultValue)` where `defaultValue` has been specified before?